src/testdrive/src/parser.rs

```diff
@@ -520,19 +520,21 @@ impl ArgMap {
         }
     }
 
+    /// Get a boolean, if it is missing return false
     pub fn opt_bool(&mut self, name: &str) -> Result<bool, String> {
-        match self.opt_string(name) {
-            Some(val) => {
-                if val == "true" {
-                    Ok(true)
-                } else if val == "false" {
-                    Ok(false)
-                } else {
-                    Err(format!("bad value for boolean parameter {}: {}", name, val))
-                }
+        self.bool_(name).unwrap_or(Ok(false))
+    }
+
+    pub fn bool_(&mut self, name: &str) -> Option<Result<bool, String>> {
+        self.opt_string(name).map(|val| {
+            if val == "true" {
+                Ok(true)
+            } else if val == "false" {
+                Ok(false)
+            } else {
+                Err(format!("bad value for boolean parameter {}: {}", name, val))
             }
-            None => Ok(false),
-        }
+        })
```

Comment:
These aren't in keeping with the pattern.
  * `opt_string -> Option<String>`
  * `string -> Result<String, String>`
  * `opt_parse -> Result<Option<T>, String>`
  * `parse -> Result<T, String>`

Granted `opt_bool` was already in violation of the rule, but this makes the situation pretty confusing. I recommend that we change `opt_bool` to return `Result<Option<bool>, String>`, and then the one existing call site can change like so:

```
let publish = cmd.args.opt_bool("publish")?.unwrap_or(false);
```